src/syntax-tree/ast.ts

```diff
@@ -1052,8 +1052,8 @@ export class ForStatement extends Statement implements VariableContainer {
 
     assignNewVariable(varController: VariableController) {
         this.assignId();
+        this.loopVar.updateIdentifier(this.getIdentifier(), this.getIdentifier(), false);
```

Comment:
This one is interesting. `updateIdentifier()` will update the identifier in the button as well, but if it is done before the button is created it tries to update a button that does not exist. This is not an issue for regular `VarAssignmentStmt` because they get their identifier updated directly as we type. However, the `VarAssignmentStmt` of a for-loop is not actually in the editor, it is only in the AST backend so when updates to the identifier of a for-loop var are made the actual statement that backs it is updated indirectly causing the order of the updates to matter.